File: src/client/views/utility-list.ts
// AUTO GENERATED
import { tsx_map } from "@wirunekaewjai/tiny-tsx/macro";

export interface UtilityListItem {
  id: string;
  icon: string;
  name: string;
}

export const $utility_list = (items: UtilityListItem[]) => `<div class="space-y-2 divide-y" hx-boost="true">${tsx_map(items, (item) => `<a class="hover:text-blue-400 grid grid-cols-[16px_1fr] gap-x-4 px-4 py-2 items-center" href="/utils/${item.id}">${item.icon}${item.name}</a>`)}</div>`;

/*
interface Item {
  id: string;
  icon: string;
  name: string;
}

(items: Item[]) => (
  <div
    class="space-y-2 divide-y"
    hx-boost="true"
  >
    {map(items, (item) => (
      <a
        class="hover:text-blue-400 grid grid-cols-[16px_1fr] gap-x-4 px-4 py-2 items-center"
        href={`/utils/${item.id}`}
      >
        {item.icon}
        {item.name}
      </a>
    ))}
  </div>
);
*/

File: src/server/views/utility_list.rs
use html_to_string_macro::html;

use crate::functions;

pub fn utility_list(items: Vec<(&str, &str, &str)>) -> String {
    return html!(
        <div
            class="space-y-2 divide-y"
            hx-boost="true"
        >
            {
                functions::map_children(&items, &|(id, icon_name, name)| html!(
                    <a
                        class="hover:text-blue-400 grid grid-cols-[16px_1fr] gap-x-4 px-4 py-2 items-center"
                        href={format!("/utils/{id}")}
                        x-component="link"
                    >
                        <i class={format!("fa-solid {icon_name}")}></i>
                        {name}
                    </a>
                ))
            }
        </div>
    );
}
